### DISCUSSION TITLE
error when moving item to vault "Organization mismatch. Please resync the client before updating the cipher"

### DISCUSSION THREAD

**Discussion by catdevnull** [dani-garcia/vaultwarden#3771] [Q&A]
### Subject of the issue

when trying to add an item to a new organization, an error that says "Organization mismatch. Please resync the client before updating the cipher" appears (like #346)

even after restarting, the same thing happens
### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.29.1
* Web-vault version: v2023.5.0
* OS/Arch: linux/x86_64
* Running within Docker: true (Base: Debian)
* Environment settings overridden: true
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Browser/Server Time Check: true
* Server/NTP Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.41.2
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** DOMAIN, ADMIN_TOKEN

```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": false,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "_smtp_img_src": "cid:",
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_session_lifetime": 20,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "***************",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://*******************",
  "domain_origin": "*****://*******************",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 3 * * * *",
  "emergency_request_timeout_schedule": "0 7 * * * *",
  "enable_db_wal": true,
  "event_cleanup_schedule": "0 10 0 * * *",
  "events_days_retain": null,
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "Sutty",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "Info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "org_events_enabled": false,
  "org_groups_enabled": false,
  "password_hints_allowed": true,
  "password_iterations": 100000,
  "push_enabled": false,
  "push_installation_id": "***",
  "push_installation_key": "***",
  "push_relay_uri": "https://push.bitwarden.com",
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sendmail_command": null,
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": false,
  "signups_domains_whitelist": "",
  "signups_verify": true,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_embed_images": true,
  "smtp_explicit_tls": true,
  "smtp_from": "*********************",
  "smtp_from_name": "Vaultwarden",
  "smtp_host": "***************",
  "smtp_password": "***",
  "smtp_port": 465,
  "smtp_security": "force_tls",
  "smtp_ssl": true,
  "smtp_timeout": 15,
  "smtp_username": "*********************",
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_sendmail": false,
  "use_syslog": false,
  "user_attachment_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": true,
  "websocket_port": 3012,
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>

* Install method: Docker

* Clients used: Web vault 

* Reverse proxy and version: Caddy 2

### Steps to reproduce

1. create a new organization
2. try to move an item from the user vault to the org vault

### Expected behaviour

it works

### Actual behaviour

it shows this error

![image](https://github.com/dani-garcia/vaultwarden/assets/86896666/cd18ca3c-6c24-4dfc-9877-6cb70d943be3)

### Troubleshooting data

```
config-vaultwarden-1  | [2023-08-13 13:02:11.351][request][INFO] PUT /api/ciphers/REDACTED UUID/share
config-vaultwarden-1  | [2023-08-13 13:02:11.358][vaultwarden::api::core::ciphers][ERROR] Organization mismatch. Please resync the client before updating the cipher
config-vaultwarden-1  | [2023-08-13 13:02:11.358][response][INFO] (put_cipher_share) PUT /api/ciphers/<uuid>/share => 400 Bad Request
```

**Comment by BlackDex** [comment]
Have you tried to logout and login again?

**Comment by catdevnull** [comment]
yes, the same error appears.

**Comment by BlackDex** [comment]
According to the error the UUID of the organization is not valid compared to the data posted.
Try to use the Browsers Dev Console (F12) and see if the Posted data, and then in particular the `organizationId` in the `cipher` tree if that matches or is correct with the organization uuid you want it to move to.

It seems to actually say that the specific cipher already is assigned to an other organization. And moving from one org to an other is not possible.

Not sure how that cipher which is in your own vault is registered in the database, but that should then contain an org uuid already. And that is what makes it break currently.

I just tried the steps, and it works just fine for me.

**Comment by catdevnull** [comment]
well, the organizationId does appear to match. 
>And moving from one org to an other is not possible.

i guess this is what's going wrong. why is it not possible? sounds like a bug

**Comment by BlackDex** [comment]
Moving from one org to an other isn't possible at all. I think  you are trying to use the `selectable` menu at the top right to just move one (or more) item(s). That will only work for personal items, not for other organizational items.

You need to clone the items instead of moving them.

So, this is by design.

**Reply by catdevnull** [reply]
well, it shouldn't be possible to do by UI and even if it is it should show an error explaining why (that's a bug)

it's also unclear to me how i would go about cloning it to another organization

**Comment by briandking** [comment]
I just ran into this same issue and this discussion was the first link to show up in google.

1. The webUI does allow you to move from Org to Org, using the menu. In the Vaults tab, select the item you want to move and use the upper right hand menu option "Move to selected Organization".
2. The move does fail 
3. with the misleading error asking you to resync your client.

If the issue is item 1 or 3, then it's not a vaultwarden issue. If the issue is number 2, then it's possibly a vaultwarden issue.

For @catdevnull (and for anyone else who discovers this the same way I did) - as for how to clone from an Organization to your personal vault, and then from your Vault to the New Organization, it can't be done from the Vault tab, it's done from the Organization Tab and it's a bit convoluted.

1. Go to the Organization tab
2. Select the source organization from the dropdown at the top
3. Select the item type or All items (search won't find it if you don't)
4. Search for the item you want to move
5. Use the 3 dot menu on the right of the item and select "Clone"
6. Scroll down to "Who owns this item" and select your personal vault (you can select the target org but it will fail because the collections list won't populate for it and you have to choose a collection -- another possible UI bug (other orgs should be listed) or a vaultwarden bug (the collections for the target org should be found))
7. Click Save
8. Go back to the Vaults tab
9. Search for the item
10. Select the cloned version (owner=me) and with the right hand 3 dot menu select "Move to Organization"
11. Select the Organization and the collections and Save

Summary of _potential_ bugs:
UI (bitwarden)

1. Allows you to attempt to move from org to org in the vault tab
2. Misleading error message when the move from org to org fails fro the vault tab
3. Allows you to attempt to move from org to org in the Organization tab
4. Doesn't show the the available collections after select Org to Org move from the Organizations tab

Vaultwarden

1. When moving from Org to Org in the vaults tab, move fails
2. When moving from Org to Org in the Org tab, listing target Org collections fails

**Reply by alibehrouzi** [reply]
Excellent summary @briandking. I too am experiencing the exact same symptoms. I'm on a self-hosted Vaultwarden (on a Synology NAS). I need to get past this so that I have my family officially start using the tool. 

Any help would be appreciated.

**Comment by BlackDex** [comment]
I would suggest people to read https://bitwarden.com/help/sharing/
The Web-UI supports moving multiple personal ciphers to an organization, not from org to org.

That the Web-UI does not validate if you have selected an organizational cipher is not something we can fix on the server side.
We can check if an attempt like this is done, and block that, since it will not work because the encryption will fail.

That you see an error message with Vaultwarden is at least more than you will see with Bitwarden, that doesn't show any error message at all, and it will also fail silently if you select a mix of personal owned and organizational owned ciphers and try to move them.

You can describe any workflow you want here the main point is. It is not possible to move ciphers from one org to an other at this point in time.

You can only move personal owned ciphers to an organization.

Until Bitwarden has created a feature like this in there client, this project can't do anything about this.
We could maybe silently ignore org ciphers and still allow personal ciphers, and maybe we can try to detect this a bit better and provide a bit more descriptive error message, but that is going to be about it.

We can't magically move the item from one org to an other on the server side, that will just not work, if you do not believe me, then just try to modify the database and change a cipher an other organization id, and see what happens with users who are not in both organizations.

There are probably a few feature requests open already at Bitwarden regarding this, i suggest to follow those.

**Reply by briandking** [reply]
My summary intention was to

1.  describe the steps the user has to follow to move from org to personal to org (because it was not all that easy to find); and
2.  highlight that the error message is not intuitive at all for an end user. "Please resync your client" had me focusing on client side issues for a long time before I found this discussion and then was able to figure out it was a workflow issue. If there is a way to make the error more informative it would greatly improve the user experience.

Thank you for making vaultwarden available and for taking the time to respond.
